Sell/Product/Tests/ProductRepositoryTests.swift

```diff
@@ -0,0 +1,164 @@
+//
+//  ProductRepositoryTests.swift
+//  RealifeTechTests
+//
+//  Created by Mickey Lee on 23/06/2021.
+//  Copyright © 2021 Realife Tech. All rights reserved.
+//
+
+import XCTest
+import Apollo
+@testable import RealifeTech
+
+final class ProductRepositoryTests: XCTestCase {
+
+    private let productFragment = ApolloType.FragmentProduct(id: "1")
+
+    private func makeGraphQLManagerAndSUT<DataType>(
+        ofType type: DataType.Type
+    ) -> (graphQLManager: MockGraphQLManager<DataType>, sut: ProductRepository) {
+        let graphQLManager = MockGraphQLManager<DataType>()
+        let sut = ProductRepository(graphQLManager: graphQLManager)
+        return (graphQLManager: graphQLManager, sut: sut)
+    }
+}
+
+extension ProductRepositoryTests {
+
+    func test_getProducts_graphQLManagerHasData_completeWithSuccessCase() {
+        let (graphQLManager, sut) = makeGraphQLManagerAndSUT(ofType: ApolloType.GetProductsQuery.Data.self)
+
+        let getProducts = ApolloType
+            .GetProductsQuery
+            .Data
+            .GetProduct(edges: [], nextPage: 123)
+        let data = ApolloType
+            .GetProductsQuery
+            .Data(getProducts: getProducts)
+        let expectedResult = GraphQLResult<
+            ApolloType
+            .GetProductsQuery
+            .Data>(
```

Comment:
This line break style makes it hard to see the difference between generic constraints and the input arguments. Is it possible to line break the `>(` onto its own line?

- [ ] Consider adding line break if possible